crate/utils/src/crypto/hybrid_encryption/ecies.rs

```diff
@@ -0,0 +1,204 @@
+use num_bigint_dig::algorithms::idiv_ceil;
+#[cfg(feature = "fips")]
+use openssl::nid::Nid;
+use openssl::{
+    bn::BigNumContext,
+    ec::{EcGroupRef, EcKey, EcPoint, EcPointRef, PointConversionForm},
+    hash::{Hasher, MessageDigest},
+    pkey::{PKey, Private, Public},
+    symm::{decrypt_aead, encrypt_aead, Cipher},
+};
+
+#[cfg(feature = "fips")]
+use crate::kmip_utils_bail;
+use crate::{
+    crypto::symmetric::{AES_256_GCM_IV_LENGTH, AES_256_GCM_KEY_LENGTH, AES_256_GCM_MAC_LENGTH},
+    error::KmipUtilsError,
+};
+
+/// Derive initialization vector from recipient public key `Q` and ephemeral
+/// public key `R` using SHAKE128-128.
```

Comment:
Concerning one-letter bindings, it just is common notation for elliptic curve points and variables. I figured that it would be easier for anyone to read after me if I use the names of variables given in algorithmic descriptions of ECIES i.e. `G`, `Q`, `R` and `S` for EC points and `d`, `r`, ... for scalars.

I'm well aware of the naming in crypto_box but from my experience of crossing this resource with some more standard ones on ECIES, I've seen myself confused to translate to which variable means what. In this sense I must support my point of the naming I chose.

I agree the crate uses `nonce` as a naming for IV in symmetric encryption but I feel this is off and `iv` should be used  hence my choice.

Reply:
OK for the use of "iv".

If you want to use the one-letter bindings, you need to give the link to the standard you are following. Readers (myself included) may not be familiar with those which really makes the reading more complicated (especially the fact that Q = dG if I understood correctly).